text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,144 @@
+2. A 56-bit trace _threshold_ as expressed in the TraceState, called `T` below.
+
+The sampling decision is propagated with the following algorithm:
+
+* If the `th` key is not specified, Always Sample.
+* Else derive `T` by parsing the `th` key as a hex value as described below.
+* If `T` is 0 and the _sampled_ flag is set, Always Sample. This implies that non-probabilistic sampling is taking place.
+* Compare the 56 bits of `T` with the 56 bits of `R`. If `T <= R`, then do not sample.
+
+The `R` value MUST be derived as follows:
+
+* If the key `rv` is present in the Tracestate header, then `R = rv`.
+* Else if the Random Trace ID Flag is `true` in the traceparent header, then `R` is the lowest-order 56 bits of the trace-id.
+* Else `R` MUST be generated as a random value in the range `(0, (2**56)-1)` and added to the Tracestate header with key `rv`.
+
+The preferred way to propagate the `R` value is as the lowest 56 bits of the trace-id.
+If these bits are in fact random, the `random` trace-flag SHOULD be set as specified in [the W3C trace context specification](https://w3c.github.io/trace-context/#trace-id).
+There are circumstances where trace-id randomness is inadequate (for example, sampling a group of traces together); in these cases, an `rv` value is required.
+
+The value of the `rv` and `th` keys MUST be expressed as up to 14 hexadecimal characters from the set `[0-9a-f]`. Trailing zeros (but not leading zeros) may be omitted.
+
+Examples:
+`th` value is missing: Always Sample (probability = 100%). The AlwaysSample sampler in the OTel SDK should do this.
+`th=8` -- equivalent to `th=80000000000000`, which is 50% probability.
+`th=08` -- equivalent to `th=08000000000000`, which is 3.125% probability.
+`th=0` -- equivalent to `th=00000000000000`, which means Always Sample; this is outside of probabalistic sampling.
+
+The `T` value MUST be derived as follows:
+
+* If the `th` key is not present in the Tracestate header, then `T` is effectively 2^56 (which doesn't fit in 56 bits).
+* Else the value corresponding to the `th` key should be interpreted as above.
+
+Sampling Decisions SHOULD be propagated by setting the value of the `th` key in the Tracestate header according to the above.
+
+## Changing T and R values
+
+The T value MAY be modified.
+
+In the case of a downstream sampler -- a tail sampler on the collection path that is attempting to reduce the volume of traffic -- the sampler MAY modify the `th` header by reducing its value.
+It MUST NOT increase it, as it is not possible to retroactively adjust the sampling probability upward.
+
+A non-root head sampler MAY raise or lower the T value.
+This may be necessary in situations where a non-root sampler wishes to ensure that the remainder of the trace is sampled due to some exceptional occurrence.
+Note that changing the probability of a trace in flight introduces inconsistency and may cause the trace to be incomplete.
+
```

Comment:
I suggest to reword the above as:
A consistent head sampler MUST set the T value corresponding to the sampling probability it actually uses. If it samples a non-root span, it MAY use the sampling probability of the parent span and use its T value.
Using different sampling probabilities for spans belonging to the same trace will lead to incomplete traces.